compiler/noirc_frontend/src/elaborator/comptime.rs

```diff
@@ -11,42 +9,40 @@ impl<'context> Elaborator<'context> {
     /// Elaborate an expression from the middle of a comptime scope.
     /// When this happens we require additional information to know
     /// what variables should be in scope.
-    pub fn elaborate_item_from_comptime<T>(
-        &mut self,
+    pub fn elaborate_item_from_comptime<'a, T>(
+        &'a mut self,
         current_function: Option<FuncId>,
-        f: impl FnOnce(&mut Self) -> T,
+        f: impl FnOnce(&mut Elaborator<'a>) -> T,
     ) -> T {
-        self.function_context.push(FunctionContext::default());
-        let old_scope = self.scopes.end_function();
-        self.scopes.start_function();
-        let function_id = current_function.map(DependencyId::Function);
-        let old_item = replace(&mut self.current_item, function_id);
+        // Create a fresh elaborator to ensure no state is changed from
+        // this elaborator
+        let mut elaborator = Elaborator::new(
+            self.interner,
+            self.def_maps,
+            self.crate_id,
+            self.debug_comptime_in_file,
+        );
```

Comment:
I didn't encounter any errors in reusing the same elaborator but I think this approach is safer and requires less tear-down code.